extractor/filesystem/os/chrome/extensions/extensions.go

```diff
@@ -0,0 +1,241 @@
+	path := api.Path()
+	path = filepath.ToSlash(path)
+
+	switch runtime.GOOS {
+	case "windows":
+		return windowsChromeExtensionsPattern.MatchString(path) || windowsChromiumExtensionsPattern.MatchString(path)
+	case "linux":
+		return linuxChromeExtensionsPattern.MatchString(path) || linuxChromiumExtensionsPattern.MatchString(path)
+	case "darwin":
+		return macosChromeExtensionsPattern.MatchString(path) || macosChromiumExtensionsPattern.MatchString(path)
+	default:
+		return false
+	}
+}
+
+// Extract extracts chrome extensions
+func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) ([]*extractor.Inventory, error) {
+	var m manifest
+	if err := json.NewDecoder(input.Reader).Decode(&m); err != nil {
+		return nil, fmt.Errorf("could not extract manifest from %s: %w", input.Path, err)
+	}
+	if err := m.validate(); err != nil {
+		return nil, fmt.Errorf("bad format in manifest %s: %w", input.Path, err)
+	}
+
+	// extract the extensions ID from the path
+	id, err := extractExtensionsIDFromPath(input)
+	if err != nil {
+		return nil, fmt.Errorf("could not extract extension id from %s: %w", input.Path, err)
+	}
+
+	// if default locale is specified some fields of the manifest may be
+	// written inside the ./_locales/LOCALE_CODE/messages.json file
+	if m.DefaultLocale != "" {
+		if err := extractLocaleInfo(&m, input); err != nil {
+			return nil, fmt.Errorf("could not extract locale info from %s: %w", input.Path, err)
+		}
+	}
+
+	ivs := []*extractor.Inventory{
+		{
+			Name:    id,
+			Version: m.Version,
+			Metadata: &Metadata{
+				AuthorEmail:          m.Author.Email,
+				Description:          m.Description,
+				HostPermissions:      m.HostPermissions,
+				ManifestVersion:      m.ManifestVersion,
+				MinimumChromeVersion: m.MinimumChromeVersion,
+				Name:                 m.Name,
+				Permissions:          m.Permissions,
+				UpdateURL:            m.UpdateURL,
+			},
+		},
+	}
+
+	return ivs, nil
+}
+
+// extractExtensionsIDFromPath extracts the extensions id from the path
+//
+// expected path is:
+//
+//	/extensionID/version/manifest.json
+func extractExtensionsIDFromPath(input *filesystem.ScanInput) (string, error) {
+	path, err := filepath.Abs(input.Path)
+	if err != nil {
+		return "", fmt.Errorf("could not extract full path: %w", err)
+	}
+	parts := strings.Split(filepath.ToSlash(path), "/")
+	if len(parts) < 3 {
+		return "", errors.New("cold not find id expected path format '/extensionID/version/manifest.json'")
+	}
+	id := parts[len(parts)-3]
+	// no more validation on the id is required since the path has been checked during FileRequired
+	return id, nil
+}
+
+// extractLocaleInfo extract locale information from the _locales/LOCALE_CODE/messages.json
+// following manifest.json v3 specification
+func extractLocaleInfo(m *manifest, input *filesystem.ScanInput) error {
+	messagePath := filepath.Join(filepath.Dir(input.Path), "/_locales/", m.DefaultLocale, "message.json")
```

Comment:
No need for the leading and ending slashes in "/_locales/" if you use filepath.Join